src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0041-View-AllStudentSchoolDim-Create.sql

```diff
@@ -0,0 +1,129 @@
+﻿-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+IF EXISTS (
+        SELECT 1
+        FROM INFORMATION_SCHEMA.VIEWS
+        WHERE TABLE_SCHEMA = 'analytics'
+            AND TABLE_NAME = 'AllStudentSchoolDim'
+        )
+BEGIN
+    DROP VIEW analytics.AllStudentSchoolDim;
+END;
+GO
+
+CREATE VIEW analytics.AllStudentSchoolDim
+AS
+    SELECT CONCAT (
+            Student.StudentUniqueId
+            ,'-'
+            ,StudentSchoolAssociation.SchoolId
+            ,'-'
+            ,CONVERT(varchar, StudentSchoolAssociation.EntryDate, 112)
+            ) AS AllStudentSchoolKey
+        ,CONCAT (
+            Student.StudentUniqueId
+            ,'-'
+            ,StudentSchoolAssociation.SchoolId
+            ) AS StudentSchoolKey
+        ,Student.StudentUniqueId AS StudentKey
+        ,CAST(StudentSchoolAssociation.SchoolId AS VARCHAR) AS SchoolKey
+        ,COALESCE(CAST(StudentSchoolAssociation.SchoolYear AS VARCHAR), 'Unknown') AS SchoolYear
+        ,Student.FirstName AS StudentFirstName
+        ,COALESCE(Student.MiddleName, '') AS StudentMiddleName
+        ,COALESCE(Student.LastSurname, '') AS StudentLastName
```

Comment:
In DS 3.x student lastname can't be null, so we can remove the `coalesce` call here. And in DS 3.2 and 3.3.

Reply:
The column was updated to remove the coalesce.